smac/optimizer/smbo.py

```diff
@@ -131,21 +127,25 @@ def __init__(self,
         self.stats = stats
         self.initial_design = initial_design
         self.runhistory = runhistory
-        self.rh2EPM = runhistory2epm
         self.intensifier = intensifier
         self.aggregate_func = aggregate_func
         self.num_run = num_run
-        self.model = model
-        self.acq_optimizer = acq_optimizer
-        self.acquisition_func = acquisition_func
         self.rng = rng
-        self.random_configuration_chooser = random_configuration_chooser
 
-        self._random_search = RandomSearch(
-            acquisition_func, self.config_space, rng
-        )
-
-        self.predict_incumbent = predict_incumbent
+        self.initial_design_configs = []
+
+        # initialize the chooser to get configurations from the EPM
+        self.epm_chooser = EPMChooser(scenario=scenario,
```

Comment:
Shouldn't this be constructed in the ACFacade?

Reply:
The other optimizers (like epils) have a different method of selecting the next challenger, so I am not sure if that would be right to pass the chooser as an argument to any general smbo class. Any recommendations?

Reply:
Hm, but this is not accounted for right now anyway, right?

Reply:
Not yet. But the main BO loop in that should change too, for the new design.

Reply:
okay, that makes sense. Thanks for pointing that out.